broadway/api/utils/multiqueue.py

```diff
@@ -0,0 +1,45 @@
+from queue import Queue, Empty
+
+"""
+A wrapper around Python's queue module that contains multiple queues
+and round-robins between them when pulling.
+"""
+
+
+class MultiQueue:
+    def __init__(self):
+        self.queues = {}
+        self.keys = []
+        self.round_robin_idx = 0
+
+    def add_queue(self, queue_id):
+        if queue_id in self.queues:
+            raise Exception("{} already exists in the MultiQueue.".format(queue_id))
+
+        self.queues[queue_id] = Queue()
+        self.keys.append(queue_id)
+
+    def push(self, queue_id, elem):
+        if queue_id not in self.queues:
+            raise Exception("{} does not exist in the MultiQueue.".format(queue_id))
+
+        self.queues[queue_id].put(elem)
+
+    def pull(self):
+        N = len(self.keys)
+        if N == 0:
+            raise Empty("MultiQueue has no queues in it.")
+
+        for i in range(N):
+            idx = (self.round_robin_idx + i) % N
+
+            try:
+                rv = self.queues[self.keys[idx]].get_nowait()
+                self.round_robin_idx = (idx + 1) % N
+                return rv
+
+            except Empty:
+                # continue to next queue
+                pass
+
+        raise Empty("All the queues in the MultiQueue are empty.")
```

Comment:
Do we want to error here? HTTP nodes try to keep polling and most of the time when they poll, the queue is empty.

Reply:
I made it raise `Empty` since currently, that's how `worker_schedule_job` figures out that there's no more jobs available. 
https://github.com/illinois-cs241/broadway/blob/4eda74540630624a9b26319af1999ddbe4c4eb98/broadway/api/callbacks/worker.py#L66-L92

Reply:
Oh, you are right. Raising an `Empty` is the correct thing to do.